R/inven.R

```diff
@@ -31,35 +28,73 @@ inven <- function(x,obj,need="@all",crump=TRUE,...) {
     stop("need can only be parameter names", call.=FALSE) 
   }
   
-  if(!all(need %in% names(obj))) {
-    if(crump) inven_stop(x,obj,need)
-    return(inven_report(x,obj,need))
+  if(crump) {
+    inven_stop(obj,need)
   }
   
-  return(invisible(TRUE))
+  return(inven_report(obj, need))
 }
 
-##' @rdname inven
-##' @export
-inventory <- function(x,obj,...) {
-  dots <- lazyeval::lazy_dots(...)
-  need <- select_vars_(names(param(x)),dots)
-  if(length(need)==0) need <- names(param(x))
-  ans <- inven(x,obj,need,crump=FALSE) 
-  if(ans) message("Found all required parameters.")
-  return(invisible(ans))
+#' Check whether all required parameters needed in a model are present in an object
+#' @param x model object
+#' @param obj dataframe to pass to \code{\link{idata_set}} or \code{\link{data_set}}
+#' @param ... capture dplyr-style parameter requirements
+#' @param .strict whether to stop execution if all requirements are present (\code{TRUE}) 
+#' or just warn (\code{FALSE}); see details
+#' @examples \dontrun{
+#' inventory(mod, idata, CL:V) # parameters defined, inclusively, CL through Volume 
+#' inventory(mod, idata, everything()) # all parameters
+#' inventory(mod, idata, contains("OCC")) # all parameters containing OCC
+#' inventory(mod, idata, -F) # all parameters except F
+#' }
+#' @return original mrgmod
+#' @details 
+#' If parameter requirements are not explicitly stated, the requirement defaults to 
+#' all parameter names in \code{x}.  Note that, by default,
+#' the inventory is not \code{.strict} by default unless the user explicitly
+#' states the parameter requirement. That is, if parameter requirements are explicitly 
+#' stated, \code{.strict} will be set to \code{TRUE} if a value \code{.strict} was not
+#' passed in the call.
+#' @export
+inventory <- function(x,obj,..., .strict = FALSE) {
+  
+  oname <- as.character(as.list(match.call())$obj)
+
+  need <- select_vars(names(param(x)),...)
+  
+  if(!length(need)) {
+    need <- names(param(x))
+  } else {
+    if(missing(.strict)) .strict <- TRUE 
```

Comment:
but that should just be handled by the default/or user provided values.


eg in the function spec just have function(,... , .strict = TRUE)

no need to check missingness, and reassign, if the user doesn't specify, then its true by default already.

That is the 'beauty' of forcing a full name match after ..., there is no real way for them to accidentally be specifying it if they didn't realize.